app/views/admin/users/widgets/_organizations.html.erb

```diff
@@ -25,21 +25,23 @@
         </h3>
       </div>
       <div class="c-card__revealable shrink-0">
-        <button type="button" class="c-btn c-btn--icon-alone" data-modal-trigger data-modal-title="Edit organization" data-modal-size="small" data-modal-content-selector="#edit-organization-<%= org_membership.organization.id %>">
+        <button type="button" class="c-btn c-btn--icon-alone" data-modal-trigger data-modal-title="Edit organization" data-modal-size="small" data-modal-content-selector="#edit-organization-<%= org_membership.organization.id %>" aria-label="Edit <%= org_membership.organization.name %> organization membership">
           <%= crayons_icon_tag(:edit, class: "c-btn__icon") %>
         </button>
         <%= form_with model: [:admin, org_membership], method: :delete, local: true, html: { class: "inline" } do |f| %>
-          <%= f.button crayons_icon_tag(:x, class: "c-btn__icon"), class: "c-btn c-btn--destructive c-btn--icon-alone", data: { confirm: "Are you sure you want to remove them from #{org_membership.organization.name}?" } %>
+          <%= f.button crayons_icon_tag(:x, class: "c-btn__icon"), class: "c-btn c-btn--destructive c-btn--icon-alone", data: { confirm: "Are you sure you want to remove them from #{org_membership.organization.name}?" },
+                                                                   aria: { label: "Revoke #{org_membership.organization.name} organization membership" } %>
         <% end %>
       </div>
     </div>
 
     <div id="edit-organization-<%= org_membership.organization.id %>" class="hidden">
       <%= form_with model: [:admin, org_membership], method: :patch, local: true do |f| %>
         <div class="crayons-field mb-4">
-          <%= f.label "Permission level", class: "crayons-field__label" %>
+          <%= f.label :type_of_user, "Permission level", class: "crayons-field__label" %>
           <p class="crayons-field__description">Users can be assigned an Admin role or can remain a regular member of this organization.</p>
-          <%= f.select("type_of_user", options_for_select(%w[member admin], selected: org_membership.type_of_user), {}, class: "crayons-select") %>
+          <%# Added the aria-label even though it's redundant because for some reason E2E tests do not pick up the label association, even though it works in browsers. %>
```

Comment:
Curious if you have any ideas why Cypress might not be picking up the label text even though the browser does @aitchiss.

Reply:
I'll take a look at this... definitely we want to understand what's going on with that!

Reply:
Ahh I didn't catch that duplicate ID issue. I'll get that sorted. Thanks!